utils/joiner/tuplejoiner.h

```diff
@@ -344,13 +361,13 @@ class TupleJoiner
     TupleJoiner();
     TupleJoiner(const TupleJoiner&);
     TupleJoiner& operator=(const TupleJoiner&);
+    void getBucketCount();
 
     rowgroup::RGData smallNullMemory;
 
-
-    boost::scoped_ptr<hash_t> h;  // used for UM joins on ints
-    boost::scoped_ptr<sthash_t> sth;  // used for UM join on ints where the backing table uses a string table
-    boost::scoped_ptr<ldhash_t> ld;  // used for UM join on long double
+    boost::scoped_array<boost::scoped_ptr<hash_t> > h;  // used for UM joins on ints
```

Comment:
This nest of boost entities attracts my attention a lot. I wonder if std::unique_ptr<std::vector<std::unique_ptr<hash_t> > > will do the trick?

Reply:
No need for the first unique_ptr even.  Yes it would do the trick, but it's not necessary right now.